### DISCUSSION TITLE
Component Request: Steps / Stepper

### DISCUSSION THREAD

**Discussion by pevey** [melt-ui/melt-ui#191] [Ideas]
I deal a lot with insurance workflows, and we often have a need for step components.  This is very similar to tabs, but with configurable constraints on moving to the next step.  

As an example, think of a basic quote workflow where you might enter name, and then address, and then something else blah blah, and then see your quote, and then click to bind. 

It is similar with claims.  People tend to like these as small, clear steps as opposed to a huge confusing forms requiring tons of info at once, without great guidance through the flow.  But they ONLY prefer it if the ergonomics/accessibility is good.  E.g., no having to take hands off the keyboard to click next if you don't want.  Tab and Enter should be enough.

Just wanted to put this out there for consideration as a component type to add to the roadmap.  Thanks.

**Comment by huntabyte** [comment]
I'm trying to think about how we could make this in a headless manner, or if it should be. I want to say something like this could be possible already with a mix of tabs, pagination, and some custom logic. We do a lot of discussion in our Discord, so feel free to pop in there and bring the question up!
